src/transformers/modeling_utils.py

```diff
@@ -377,7 +381,11 @@ def shard_checkpoint(
             sharded_state_dicts[block_id][key] = weight
             continue
 
-        weight_size = weight.numel() * dtype_byte_size(weight.dtype)
+        if weight.__class__.__name__ in ["QTensor", "QBitsTensor"]:
+            # we don't add the scales/zero-point because those can be negligeable + subject to changes
```

Comment:
This means we are stuck with this if else for FC? if things are subject to change I don't really think we should have a if else here!

Reply:
> if things are subject to change I don't really think we should have a if else here!

Oh, it mean that it can change for the _scales and the _zeropoint, so I decided to not add them. For _data, it shouldn't change. I will try to find a better solution. This is not pretty indeed.

Reply:
Let me know if this [solution](https://github.com/huggingface/transformers/pull/29023/commits/15f7a2a9b9a13d0efd9736b0186994c286f73490) is enough. We shouldn't have any FC issues.